src/details/ArborX_Predicates.hpp

```diff
@@ -49,12 +49,19 @@ struct Nearest
   KOKKOS_DEFAULTED_FUNCTION
   Nearest() = default;
 
-  KOKKOS_INLINE_FUNCTION
+  KOKKOS_FUNCTION
   Nearest(Geometry const &geometry, int k)
       : _geometry(geometry)
       , _k(k)
   {}
 
+  template <class Other>
+  KOKKOS_FUNCTION auto distance(Other const &other) const
+  {
+    using Details::distance;
+    return distance(_geometry, other);
+  }
```

Comment:
```suggestion
  template <class OtherGeometry>
  KOKKOS_FUNCTION auto distance(OtherGeometry const &geometry) const
  {
    using Details::distance;
    return distance(_geometry, geometry);
  }
```

Reply:
Can do.  I was going for consistency with https://github.com/arborx/ArborX/blob/843553c9d6ba62700bc9d711a7aec0e0631fea77/src/details/ArborX_Predicates.hpp#L73-L78

Want me to update it all?

Reply:
Please.

Reply:
Done